### PR TITLE
Add rviz plugin for costmap_cspace_msgs::CSpace3D

### PR DESCRIPTION
**Pull request by nhatao**
https://user-images.githubusercontent.com/8833040/179904996-edbd7290-ef38-4090-8422-b9b965cab2b9.mp4

This plugin shows `costmap_cspace_msgs::CSpace3D` and `costmap_cspace_msgs::CSpace3DUpdate`, and is based on MapDisplay in the default rviz plugins.
https://github.com/ros-visualization/rviz/blob/fb8f3063bdc5b0aa2584cd2067c7a8bbe5e6dd02/src/rviz/default_plugin/map_display.h
https://github.com/ros-visualization/rviz/blob/fb8f3063bdc5b0aa2584cd2067c7a8bbe5e6dd02/src/rviz/default_plugin/map_display.cpp

The differences between this plugin and the original MapDisplay are following.
-  It subscribes `costmap_cspace_msgs::CSpace3D` and `costmap_cspace_msgs::CSpace3DUpdate` instead of `nav_msgs::OccupancyGrid` and `nav_msgs::OccupancyGridUpdate`.
- "Angle" property is added to show costs in a specific angle.
- "map" color scheme is removed, and the default color scheme is changed to "costmap".
- In "costmap" color scheme, "inscribed obstacle value" (drawn in cyan) is removed as it is not a special value in costmap_cspace.

### CODE REVIEW

File: costmap_cspace_rviz_plugins/include/costmap_cspace_rviz_plugins/c_space_3d_display.h [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+  /** @brief Copy msg into current_map_ and call showMap(). */
+  void incomingMap(const costmap_cspace_msgs::CSpace3D::ConstPtr& msg);
+
+  /** @brief Copy update's data into current_map_ and call showMap(). */
+  void incomingUpdate(const costmap_cspace_msgs::CSpace3DUpdate::ConstPtr& update);
+  void clear();
+  void createSwatches();
+
+  std::vector<Swatch*> swatches;
```
[the remark this answers is not in the document]
Comment (nhatao) [reply costmap_cspace_rviz_plugins/include/costmap_cspace_rviz_plugins/c_space_3d_display.h]: Done

File: costmap_cspace_rviz_plugins/package.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>costmap_cspace_rviz_plugins</name>
+  <version>0.3.1</version>
+  <description>Rviz plugins for costmap_cspace_msgs</description>
+  <maintainer email="atsushi.w@ieee.org">Atsushi Watanabe</maintainer>
+  <license>BSD</license>
+  <license>Creative Commons</license>
```
Comment (at-wat) [inline costmap_cspace_rviz_plugins/package.xml:8]: Creative Commons license in trajectory_tracker_rviz_plugins is for the icon image at https://github.com/at-wat/neonavigation_rviz_plugins/tree/master/trajectory_tracker_rviz_plugins/icons/classes
As this plugin doesn't use icon, it's not needed.

File: costmap_cspace_rviz_plugins/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+else()
+  message(STATUS "Using Qt5 based on the rviz_QT_VERSION: ${rviz_QT_VERSION}")
+  find_package(Qt5 ${rviz_QT_VERSION} EXACT REQUIRED Core Widgets)
+  set(QT_LIBRARIES Qt5::Widgets)
+endif()
+
+add_definitions(-DQT_NO_KEYWORDS)
+
+set(SRC_FILES
+  include/costmap_cspace_rviz_plugins/c_space_3d_display.h
```
Comment (at-wat) [inline costmap_cspace_rviz_plugins/CMakeLists.txt:38]: I'm not very sure this is required.
I think cmake automatically detects dependencies to static header files.

File: costmap_cspace_rviz_plugins/package.xml (hunk 2 above)
Comment (nhatao) [reply costmap_cspace_rviz_plugins/package.xml]: Removed

File: costmap_cspace_rviz_plugins/CMakeLists.txt (hunk 3 above)
Comment (nhatao) [reply costmap_cspace_rviz_plugins/CMakeLists.txt:38]: It is needed to pass `c_space_3d_display.h` to `add_library`. It seems that moc of QT builds additional C++ source file from  headers.
I added `HEADER_FILES` variable to split headers from sources.

Comment (at-wat) [reply costmap_cspace_rviz_plugins/CMakeLists.txt:38]: Ah yes, generated files aren't automatically detected by cmake.
In trajectory_tracker_rviz_plugins, I used `qtN_wrap_cpp` to explicitly handle generated cpp files.
https://github.com/at-wat/neonavigation_rviz_plugins/blob/75a9759cbc7057fe62198b00751af15a74b17713/trajectory_tracker_rviz_plugins/CMakeLists.txt#L26-L52

Comment (nhatao) [reply costmap_cspace_rviz_plugins/CMakeLists.txt]: Changed to use qt4/qt5_wrap_cpp.

**Review by at-wat — APPROVED** [review]
LGTM

**Pull request merged**